pinot-core/src/main/java/org/apache/pinot/core/operator/filter/ScanBasedFilterOperator.java

```diff
@@ -43,6 +44,9 @@ public class ScanBasedFilterOperator extends BaseFilterOperator {
     _dataSource = dataSource;
     _numDocs = numDocs;
     _nullHandlingEnabled = nullHandlingEnabled;
+    Preconditions.checkState(_dataSource.getForwardIndex() != null,
+        "Forward index disabled for column: %s, creating ScanDocIdSet unsupported!",
```

Comment:
(nit) a better message may be -- `creating ScanDocIdSet unsupported -> scan based filtering not supported`

Reply:
done